### ISSUE TITLE
RQ: Raw Query Improvement Epic

### ISSUE THREAD

**Issue by matthewmueller** [prisma/prisma#12367]
**Problem**

Raw query support provides an important escape hatch for those instances when Prisma doesn't quite support what you need it to support. Over time, we've accumulated a number of raw query bugs. We'd like to fix these issues and start addressing them in a batch.

**Goal**

We do another pass over our raw query bugs. We understand and fix the ones we can. For the ones we can't, we document the limitation and look for ways to fix them in the future.

## Tasks

- [x] https://github.com/prisma/prisma/issues/12368
- [x] https://github.com/prisma/prisma/issues/12369

## Umbrella Issues

- [ ] #12795
- [ ] #12796
- [ ] #12797

## Issues

These issues came from the [Bug Cluster Column](https://github.com/orgs/prisma/projects/60#column-17266903)

### Confirmed Bugs

Legend:
🔴 - Blocked
🟡 - Fixable but breaking
🟢 - Fixable
🟤 - Unknown

#### Float input/output
🔴 Blocked by Float rework
- [ ] https://github.com/prisma/prisma/issues/3665
- [ ] https://github.com/prisma/prisma/issues/9195

#### Incorrect QE query parameter coercion
🟢 Fixable
- [x] https://github.com/prisma/prisma/issues/8121 (bigint)
- [x] https://github.com/prisma/prisma/issues/11834 (buffer)

#### Incorrect Client result coercion
🟡 **BREAKING**: Rework needed to propagate type information from QE to Client
- [x] https://github.com/prisma/prisma/issues/9163 (decimal)
- [x] https://github.com/prisma/prisma/issues/7395 (bigint)
- [x] https://github.com/prisma/prisma/issues/12551 (bigint)
- [x] https://github.com/prisma/prisma/issues/12583 (bigint)
- [x] https://github.com/prisma/prisma/issues/4569

#### Enum input/output
🟤 Need to be discussed, not sure if there's a solution
- [ ] https://github.com/prisma/prisma/issues/9877

#### Geography type input
🔴 Blocked _but a workaround was provided_
- [x] https://github.com/prisma/prisma/issues/2847

#### Void output (non existing type)
🔴 Blocked by rust pg driver (according to the issue comments, to be confirmed)
- [ ] https://github.com/prisma/prisma/issues/3530

#### Date output
🟤 We currently ignore the timezone when deserializing `TimeTz` type. Next step: figure out why
- [ ] https://github.com/prisma/prisma/issues/7916

#### Unsupported native database type
🔴 Blocked by native database type support
- [ ] https://github.com/prisma/prisma/issues/10448
- [ ] https://github.com/prisma/prisma/issues/12344
- [ ] https://github.com/prisma/prisma/issues/11339 (array)
- [ ] https://github.com/prisma/prisma/issues/9933 (Point type)
- [ ] https://github.com/prisma/prisma/issues/12864

#### Incorrect PG query parameter type inferring
🟡 **BREAKING**: Rework needed to pass in the correct query parameters types
- [x] https://github.com/prisma/prisma/issues/10424
- [x] https://github.com/prisma/prisma/issues/4647 (interval)
- [x] https://github.com/prisma/prisma/issues/7839
- [x] https://github.com/prisma/prisma/issues/6818
- [x] https://github.com/prisma/prisma/issues/9949
- [x] https://github.com/prisma/prisma/issues/9197
- [x] https://github.com/prisma/prisma/issues/7799
- [x] https://github.com/prisma/prisma/issues/7194
- [x] https://github.com/prisma/prisma/issues/9333

#### Needs text protocol support
🔴 Blocked by Text protocol support
- [ ] https://github.com/prisma/prisma/issues/5791

#### Wrong parameters
🟢 Fixable
- [ ] https://github.com/prisma/prisma/issues/11233 (`Prisma.empty`) (minor impact)

#### Stored procedure output
⚠️ Still needs to be investigated
- [ ] https://github.com/prisma/prisma/issues/11471 (sql server) (**cannot reproduced**)
- [x] https://github.com/prisma/prisma/issues/9868
- [ ] https://github.com/prisma/prisma/issues/8206 (**unconfirmed**)

#### SQL prepared statements (wrongly escaped expressions)
🟢 Fixable - Most of those aren't bugs, users just misunderstand prepared statements.
- [x] https://github.com/prisma/prisma/issues/12456
- [ ] https://github.com/prisma/prisma/issues/11454
- [ ] https://github.com/prisma/prisma/issues/9055 (**unconfirmed**)
- [x] https://github.com/prisma/prisma/issues/10051
- [x] https://github.com/prisma/prisma/issues/12531

#### No error code

- [ ] https://github.com/prisma/prisma/issues/10141

### Unconfirmed Bugs

🟤 We can hardly reproduce those issues or they're unconfirmed
- [ ] https://github.com/prisma/prisma/issues/11532 (too many query parameters, not sure we can do anything)
- [ ] https://github.com/prisma/prisma/issues/11508 (unknown MySQL function called, can't reproduce)
- [ ] https://github.com/prisma/prisma/issues/10224 (No schema available, can't reproduce)
- [x] https://github.com/prisma/prisma/issues/7510
- [ ] https://github.com/prisma/prisma/issues/9326 (insert batch of data using raw)
- [ ] https://github.com/prisma/prisma/issues/12641 (**duplicate of 10224 - needs investigation**)

### New feature

Not really a bug, it is something we don't support (yet)

- https://github.com/prisma/prisma/issues/11582 (schemaless prisma)
- https://github.com/prisma/prisma/issues/10230 (display snippet)
- https://github.com/prisma/prisma/issues/5560
- https://github.com/prisma/prisma/issues/4327
- https://github.com/prisma/prisma/issues/3991
- https://github.com/prisma/prisma/issues/3551
- https://github.com/prisma/prisma/issues/3550
- https://github.com/prisma/prisma/issues/3337
- https://github.com/prisma/prisma/issues/5058
- https://github.com/prisma/prisma/issues/2880
- https://github.com/prisma/prisma/issues/2868

**Comment by idolize** [comment]
I've updated the issue #11454 with a repo showing a basic case of how to reproduce the issue: https://github.com/idolize/prisma-issue-11454
